### PR TITLE
Fix joint positions in URDF CLI export

### PR DESCRIPTION
**Pull request by lukicdarkoo**
I missed some steps from the `preFinalize` stage and translation fields have never been updated. It should be fine now:
image

### CODE REVIEW

**Review by omichel — APPROVED** [review]

**Comment by stefaniapedrazzi** [comment]
I'm reviewing this PR and I have some comments. I will be done soon..

File: src/webots/nodes/utils/WbNodeUtilities.cpp [hunk 1]
```diff
@@ -819,7 +819,7 @@ WbRobot *WbNodeUtilities::findRobotAncestor(const WbNode *node) {
   if (!node)
     return NULL;
 
-  while (node->parentNode()) {
+  while (node) {
```
Comment (stefaniapedrazzi) [inline src/webots/nodes/utils/WbNodeUtilities.cpp:822]: After this change, the function name is no longer consistent with the functionality.

I didn't check why this change is needed but previously the check if the node was a robot was done outside this function. So we should be consistent and keep the function as it was before, or change the name of the function and adapt the previous calls of this function.

**Review by stefaniapedrazzi — COMMENTED** [review]
I have a couple of comments and change requests on this PR.

File: src/webots/maths/WbVector3.hpp [hunk 2]
```diff
@@ -222,7 +222,7 @@ class WbVector3 {
   }
 
   // text conversion
-  QString toString(WbPrecision::Level level) const {
+  QString toString(WbPrecision::Level level = WbPrecision::DOUBLE_MAX) const {
```
[the remark this answers is not in the document]
Comment (lukicdarkoo) [reply src/webots/maths/WbVector3.hpp:225]: I am not sure what is a good approach. It's annoying when debugging. I believe it is important to have the option to change the precision, but having a default one is handy as very often we don't care about the precision.

File: src/webots/nodes/utils/WbNodeUtilities.cpp (hunk 1 above)
Comment (lukicdarkoo) [reply src/webots/nodes/utils/WbNodeUtilities.cpp:822]: Without the fix, if the robot node is the root (it always used to be Group) this function fails to find the robot ancestor.

But you are right, maybe I should detect that edge case instead of doing a quick fix.

Comment (lukicdarkoo) [reply src/webots/nodes/utils/WbNodeUtilities.cpp:822]: Actually, it doesn't change the behavior, but only fixes the edge case. The behavior and the function don't match in any case (if I am not missing something). If the passed node is a robot it will return the node itself.

Comment (stefaniapedrazzi) [reply src/webots/nodes/utils/WbNodeUtilities.cpp:822]: It changes the behavior because if the node is robot but doesn't have any robot ancestor previously it was returning NULL and now it returns the node.
A function that searches for ancestors (as described by the name) is not meant to detect if the node is a robot or not. Note that this check is almost always done externally before calling this function.

Comment (lukicdarkoo) [reply src/webots/nodes/utils/WbNodeUtilities.cpp:822]: Oh, I see, I thought you were pointing at something else

File: src/webots/maths/WbVector3.hpp (hunk 2 above)
[the remark this answers is not in the document]
Comment (stefaniapedrazzi) [reply src/webots/maths/WbVector3.hpp:225]: Ok, but I'm worried that we will always use the default option without thinking about which one is the best option to use.

File: src/webots/nodes/utils/WbNodeUtilities.cpp (hunk 1 above)
Comment (stefaniapedrazzi) [reply src/webots/nodes/utils/WbNodeUtilities.cpp:822]: Actually, I'm wrong. This is only the condition that assumes that every nodes as a parent (top parent is Group). So the behavior is unchanged.

File: src/webots/maths/WbVector3.hpp (hunk 2 above)
[the remark this answers is not in the document]
Comment (lukicdarkoo) [reply src/webots/maths/WbVector3.hpp:225]: That would be a problem if default precision was low, but since it is `DOUBLE_MAX` it is not harmful.
